Dalamud/ServiceManager.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Reflection;
+using Dalamud.Logging.Internal;
+
+namespace Dalamud
+{
+    internal static class ServiceManager
+    {
+        /// <summary>
+        /// Static log facility for Service{T}, to avoid duplicate instances for different types.
+        /// </summary>
+        public static readonly ModuleLog Log = new("SVC");
+
+        /// <summary>
+        /// Instance of <see cref="Tag"/>.
+        /// </summary>
+        public static readonly Tag TagInstance = new Tag();
+
+        /// <summary>
+        /// Kicks off construction of services that can handle early loading.
+        /// </summary>
+        public static void InitializeEarlyLoadableServices()
+        {
+            var service = typeof(Service<>);
+            foreach (var serviceType in Assembly.GetExecutingAssembly().GetTypes())
+            {
+                if (serviceType.GetInterface(nameof(IEarlyLoadableServiceObject)) == null)
+                    continue;
+
+                Log.Debug("Found early loadable service: {0}", serviceType.Name);
+                service.MakeGenericType(serviceType).InvokeMember(
+                    "Initialize",
+                    BindingFlags.InvokeMethod | BindingFlags.Static | BindingFlags.Public,
+                    null,
+                    null,
+                    null);
+            }
+        }
+
+        /// <summary>
+        /// Tag class to identify constructors to be called from Dalamud core.
+        /// </summary>
+        public class Tag
```

Comment:
Is there a reason this isn't an attribute with `AttributeTargets.Constructor`?

Reply:
Parameterless constructors may be mistakenly called for whatever reason, but this marks that you aren't supposed to construct something manually using that service initializing constructor.